### PR TITLE
feat: support search bar on star tab of user profile.

### PR DESCRIPTION
**Pull request by appleboy**
Update the repo format of star tab of user profile.

screen shot 2017-02-13 at 9 23 07 am

Origin: https://try.gitea.io/appleboy?tab=stars

screen shot 2017-02-13 at 9 25 07 am

### CODE REVIEW

**Comment by lunny** [comment]
LGTM

File: models/repo.go [hunk 1]
```diff
@@ -1830,10 +1840,22 @@ func SearchRepositoryByName(opts *SearchRepoOptions) (repos []*Repository, _ int
...
-	return repos, count, sess.
+	if err = sess.
 		Limit(opts.PageSize, (opts.Page-1)*opts.PageSize).
 		OrderBy(opts.OrderBy).
-		Find(&repos)
+		Find(&repos); err != nil {
+		return nil, 0, fmt.Errorf("Repo: %v", err)
+	}
+
+	if opts.Starred {
```
Comment (lunny) [inline models/repo.go:1850]: Maybe you could see https://github.com/go-gitea/gitea/blob/master/models/repo.go#L2156

**Review by lunny — CHANGES_REQUESTED** [review]

File: models/repo.go (hunk 1 above)
Comment (appleboy) [reply models/repo.go:1850]: https://github.com/go-gitea/gitea/pull/917/commits/34de9e9ab44b4286722662a1ef3119deae814145

**Review by lunny — APPROVED** [review]

File: models/repo.go [hunk 2]
```diff
@@ -1795,9 +1797,17 @@ func SearchRepositoryByName(opts *SearchRepoOptions) (repos []*Repository, _ int
 
 	repos = make([]*Repository, 0, opts.PageSize)
 
+	if opts.Starred && opts.OwnerID > 0 {
+		sess = x.
+			Join("INNER", "star", "star.repo_id = repository.id").
+			Where("star.uid = ?", opts.OwnerID).
+			And("LOWER(lower_name) LIKE ?", "%"+opts.Keyword+"%")
```
Comment (tboerger) [inline models/repo.go]: Need lower here? Sounds like it should already be lower

**Comment by tboerger** [comment]
I have not tested it but beside the single comment LGTM

File: models/repo.go (hunk 2 above)
Comment (lunny) [reply models/repo.go]: ^

Comment (appleboy) [reply models/repo.go]: Done

Comment (appleboy) [reply models/repo.go]: Waiting testing.

**Comment by lunny** [comment]
let L-G-T-M work

**Pull request merged**